src/analysis/asymptotics.lean

```diff
@@ -787,11 +794,32 @@ have eq₃ : is_O f (λ x, f x / g x * g x) l,
   end,
 eq₃.trans_is_o eq₂
 
-theorem is_o_iff_tendsto {f g : α → β} {l : filter α}
+theorem is_o_iff_tendsto {f g : α → 𝕜} {l : filter α}
     (hgf : ∀ x, g x = 0 → f x = 0) :
   is_o f g l ↔ tendsto (λ x, f x / (g x)) l (𝓝 0) :=
 iff.intro tendsto_nhds_zero_of_is_o (is_o_of_tendsto hgf)
 
+theorem is_o_pow_pow {m n : ℕ} (h : m < n) :
+  is_o (λ(x : 𝕜), x^n) (λx, x^m) (𝓝 0) :=
+begin
+  let p := n - m,
+  have p_pos : 0 < p := nat.sub_pos_of_lt h,
+  have : n = m + p := (nat.add_sub_cancel' (le_of_lt h)).symm,
+  simp [this, pow_add],
```

Comment:
```suggestion
  simp only [this, pow_add],
```
This proof looks like it took a fight. I briefly tried to clean it up, without much luck. Ended up with this, which isn't a huge improvement. Not suggesting you change it, just commenting that this takes an unfortunate amount of effort.

```lean
  have hn : 0 < n := (lt_of_le_of_lt (nat.zero_le _) h),
  have hnm : 0 < n - m := nat.sub_pos_of_lt h,
  rw is_o_iff_tendsto,
  { convert (continuous_pow (n - m)).tendsto (0 : 𝕜),
    { ext,
      by_cases hx : x = 0,
      { simp [hx, hn, hnm, zero_pow] }, 
      { rw [div_eq_iff_mul_eq, ←pow_add, nat.sub_add_cancel],
        { exact le_of_lt h },
        { exact pow_ne_zero _ hx } } },
    { rw zero_pow hnm } },
  { intros x hx,
    rw [pow_eq_zero hx, zero_pow hn] }
```

Reply:
I have golfed my proof a little bit, now it looks almost reasonable. I really wanted to avoid the case distinction `x = 0` and division in the proof, as this is the main advantage of `is_o`.

Reply:
I agree, but I also dislike having to introduce the `* 1` with simp. It definitely looks better now though!